src/main/python/run_regression.py

```diff
@@ -0,0 +1,184 @@
+          print(line)
+  print(OKBLUE, '='*10, 'Verifying Index Succeed', '='*10, ENDC)
+
+def construct_ranking_command(yaml_data, build_index=True):
+  """Construct the Anserini ranking commands for regression test
+  Args:
+      yaml_data (dict): The yaml config read from config file.
+      build_index (bool): If the index is not built by this script then read the index path for config
+
+  Returns:
+      (:obj:`list` of :obj:`list` of :obj:`str`):
+      The ranking commands as several commands that can be run by calling subprocess.call(command)
+  """
+  ranking_commands = [
+      [
+          os.path.join(yaml_data['root'], yaml_data['search_command']),
+          '-topicreader', yaml_data['topic_reader'],
+          '-index', os.path.join(yaml_data['index_root'] if yaml_data['index_root'] else '',
+          'lucene-index.{0}.pos+docvectors{1}'.format(yaml_data['name'], '+rawdocs' if 'storeRawdocs' in yaml_data['index_options'] else ''))
+            if build_index else yaml_data['index_path'],
+          ' '.join(model['params']),
+          '-topics', os.path.join(yaml_data['root'], yaml_data['topic_root'], topic['path']),
+          '-output', 'run.{0}.{1}.{2}'.format(yaml_data['name'], model['name'], topic['path'])
+      ]
+      for (model, topic) in list(itertools.product(yaml_data['models'], yaml_data['topics']))
+  ]
+  return ranking_commands
+
+def eval_n_verify(yaml_data, dry_run):
+  """Evaluate the ranking files and verify the results with what are stored in yaml file
+  Args:
+      yaml_data (dict): The yaml config read from config file.
+      dry_run (bool): If True, we just print out the commands without actually running them
+  """
+  print('='*10, 'Verifying Results', '='*10)
+  all_eval_cmds = []
+  try:
+      for model in yaml_data['models']:
+          for i, topic in enumerate(yaml_data['topics']):
+              for eval in yaml_data['evals']:
+                  eval_cmd = [
+                      os.path.join(yaml_data['root'], eval['command']),
+                      ' '.join(eval['params']) if eval['params'] else '',
+                      os.path.join(yaml_data['root'], yaml_data['qrels_root'], topic['qrel']),
+                      'run.{0}.{1}.{2}'.format(yaml_data['name'], model['name'], topic['path'])
+                  ]
+                  if dry_run:
+                      print(' '.join(eval_cmd))
+                      continue
+                  all_eval_cmds.append(eval_cmd)
+
+                  out = [line for line in check_output(' '.join(eval_cmd)).decode('utf-8').split('\n') if line.strip()][-1]
+                  if not out.strip():
+                      continue
+                  eval_out = out.strip().split(eval['separator'])[eval['parse_index']]
+                  expected = round(model['results'][eval['metric']][i], eval['metric_precision'])
+                  real = round(float(eval_out), eval['metric_precision'])
+                  if expected == real:
```

Comment:
To be more precise, we shouldn't use `==` to compare floating point numbers due to limited precision. `round(2.675, 2)` gives `2.67` instead of the expected `2.68`.

Python 3 already has built-in functions for testing approximate equality for floating point numbers: https://docs.python.org/3/whatsnew/3.5.html#pep-485-a-function-for-testing-approximate-equality

For Python 2, we can write a custom one: https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python